R/FilterStates.R

```diff
@@ -118,11 +97,11 @@ FilterStates <- R6::R6Class( # nolint
     #' Filter call
     #'
     #' Builds \emph{subset expression} from condition calls stored in `FilterState`
-    #' objects selection. The `lhs` of the expression is `private$output_dataname`.
-    #' The `rhs` is a call to `self$get_fun()` with `private$input_dataname`
+    #' objects selection. The `lhs` of the expression is `private$dataname`.
+    #' The `rhs` is a call to `self$get_fun()` with `private$dataname`
     #' as argument and a list of condition calls from `FilterState` objects
     #' stored in `private$state_list`.
-    #' If `input_dataname` is the same as `output_dataname` and no filters are applied,
+    #' If no filters are applied,
```

Comment:
@gogonzo maybe another issue but on line 109 do you think we could do the following:

get_call = function(call_dataname = character(0)) {
  if (length(call_dataname) == 0) {
    call_dataname <- private$dataname
  } 
  checkmate::assert_character(call_dataname)  
  // replace private$dataname with call_dataname in the rest of this function
},

Reply:
I see this is likley to be too hard